tests/test_requests.py

```diff
@@ -1,17 +1,32 @@
 import sys
-from typing import List, Optional
+import typing
+from typing import Callable, Dict, List, Optional
 
 import anyio
 import pytest
 
 from starlette.datastructures import Address, State
 from starlette.requests import ClientDisconnect, Request
 from starlette.responses import JSONResponse, PlainTextResponse, Response
-from starlette.types import Message, Scope
+from starlette.testclient import TestClient
+from starlette.types import (
+    Message,
+    Receive,
+    Scope,
+    Send,
+)
+
+TestClientFactory = Callable[..., TestClient]
 
 
-def test_request_url(test_client_factory):
-    async def app(scope, receive, send):
+def test_request_url(test_client_factory: typing.Callable[..., TestClient]) -> None:
+    async def app(
+        scope: typing.Dict[str, typing.Any],
+        receive: typing.Callable[[], typing.Awaitable[typing.Dict[str, typing.Any]]],
+        send: typing.Callable[
+            [typing.MutableMapping[str, typing.Any]], typing.Awaitable[None]
+        ],
```

Comment:
Why are you not using `Scope`, `Receive`, and `Send`?

Reply:
Updated those in all places in this file.

Just a small note:
There are some places where I have added: `test_client_factory: typing.Any`. In those places I get an error when i try to use the `TestClientFactory`:
example: `Argument "data" to "post" of "TestClient" has incompatible type "str"; expected "Optional[Mapping[str, Union[str, Iterable[str]]]]"`